### ISSUE TITLE
Cant access assets on static folder while on develop mode.

### ISSUE THREAD

**Issue by FidelVe** [gatsbyjs/gatsby#18213]
## Description

I have some html files in the static folder, if I run `gatsby build && gatsby serve` I can access these pages without trouble, but when I am on develop mode if I try to access the pages I get redirected to the 404 page.

this is the content of the static folder

```shell
static
└── freecodecamp
    └── responsive-web-design
        ├── build-a-product-landing-page
        │   ├── without-test-module
        │   │   ├── images
        │   │   │   ├── background-image-1.jpg
        │   │   │   ├── background-image-2.jpg
        │   │   │   ├── black-guitar-ver.png
        │   │   │   ├── white-guitar.svg
        │   │   │   └── white-guitar-ver.png
        │   │   ├── index.html
        │   │   └── style.css
        │   └── with-test-module
        │       ├── images
        │       │   ├── background-image-1.jpg
        │       │   ├── background-image-2.jpg
        │       │   ├── black-guitar-ver.png
        │       │   ├── white-guitar.svg
        │       │   └── white-guitar-ver.png
        │       ├── index.html
        │       └── style.css
        ├── build-a-survey-form
        │   ├── without-test-module
        │   │   ├── images
        │   │   │   ├── freecodecamp.png
        │   │   │   └── header_background.png
        │   │   ├── index.html
        │   │   └── style.css
        │   └── with-test-module
        │       ├── images
        │       │   ├── freecodecamp.png
        │       │   └── header_background.png
        │       ├── index.html
        │       └── style.css
        └── build-a-tribute-page
            ├── without-test-module
            │   ├── index.html
            │   └── style.css
            └── with-test-module
                ├── index.html
                └── style.css

15 directories, 26 files

```

### Environment

```shell
  System:
    OS: Linux 5.0 Ubuntu 19.04 (Disco Dingo)
    CPU: (4) x64 Intel(R) Core(TM) i3-8145U CPU @ 2.10GHz
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 10.15.2 - /usr/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 76.0.3809.132
    Firefox: 69.0.1
  npmPackages:
    gatsby: ^2.15.28 => 2.15.28
    gatsby-image: ^2.2.23 => 2.2.23
    gatsby-plugin-manifest: ^2.2.20 => 2.2.20
    gatsby-plugin-offline: ^3.0.11 => 3.0.11
    gatsby-plugin-react-helmet: ^3.1.10 => 3.1.10
    gatsby-plugin-sharp: ^2.2.27 => 2.2.27
    gatsby-source-filesystem: ^2.1.28 => 2.1.28
    gatsby-transformer-sharp: ^2.2.19 => 2.2.19
  npmGlobalPackages:
    gatsby-cli: 2.7.54
```

**Comment by jonniebigodes** [comment]
@FidelVe see my comment [here](https://github.com/gatsbyjs/gatsby/issues/17761#issuecomment-533816520) and the following reproduction [here](https://github.com/jonniebigodes/gatsby_iframe_example) to address a similar issue like the one you're experiencing.

**Comment by FidelVe** [comment]
Hi! @jonniebigodes 

thanks, for the reply. I tried the solution proposed on your link, and although it seems to work at first, it actually introduces other unexpected behaviors.

```javascript
const express= require('express');

exports.onCreateDevServer=({app})=>{
    app.use(express.static('public'))
}
```

when adding that code to `gatsby-node.js` any change I made to the sources while running `gatsby develop` seemed to not be updating, like the **hot reload** wasn't working.

It seems that the public folder is being served as is and was not being updated while in develop mode.

I deleted all the content inside the **public** folder to force an update and now it looks like it's working like it should.

again, thank you for the reply, although this is not a perfect solution, it gets the job done but I need to be on the lookout for any weird behavior or bug.
